### PR TITLE
Update docs links to new format in code docs and reference guides

### PR DESCRIPTION
**Pull request by bethanyaconnor**
I used the logic in #39839 and modified it a bit to bulk update these links. I am not sure I got quite all of them, but I can do another quick check to see if there are any I missed. This gets a bulk of them through.

For posterity, the code I used to update these:
```
require_relative '../../dashboard/config/environment'

BLOCK_RE = /`(?:(?<ide>\w*)\/)?(?<block>[^\(]*)(?<args>\([^\)]*\))?`/

def replace_blocks(text)
  text.gsub(/(?<!\[)`[^`]*`/) do |content|
    match = BLOCK_RE.match(content)
    if match.present?
      query = ProgrammingExpression.joins(:programming_environment)
      query = query.where(
        programming_environment: ProgrammingEnvironment.find_by(name: match[:ide])
      ) if match[:ide]
      query = query.where(name: match[:block])

      if query.count == 1
        pe = query.first
        puts pe.inspect
        puts pe.get_color
        block = "`#{pe.name}#{match[:args] || ""}`"
        block += "(#{pe.get_color})" if pe.get_color
        result = pe.documentation_path ? "[#{block}](#{pe.documentation_path})" : block
        result
      else
        content
      end
    else
      content
    end
  end
end

def update_programming_expressions
  ProgrammingExpression.all.each do |exp|
    exp.short_description = replace_blocks(exp.short_description) if exp.short_description
    exp.content = replace_blocks(exp.content) if exp.content
    exp.tips = replace_blocks(exp.tips) if exp.tips
    (exp.examples || []).each do |example|
      example['description'] = replace_blocks(example['description']) if example['description']
    end
    (exp.palette_params || []).each do |param|
      param['description'] = replace_blocks(param['description']) if param['description']
    end
    if exp.changed?
      exp.save!
      exp.write_serialization
    end
  end
end

def update_reference_guides
  ReferenceGuide.all.each do |guide|
    guide.content = replace_blocks(guide.content) if guide.content
    if guide.changed?
      guide.save!
      guide.write_serialization
    end
  end
end

update_programming_expressions
update_reference_guides
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/programming_expressions/applab/accelerometer.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
+  "content": "The [`accelerometer`(#f78183)](/docs/applab/accelerometer/) is an object representing the accelerometer sensor attached to the Circuit Playground.\r\n\r\n### Properties and Methods\r\n\r\n* [`accelerometer.getOrientation()`(#f78183)](/docs/applab/accelerometer.getOrientation/) - Get the orientation of the board (i.e tilting forward, back, side to side, spinning, etc)\r\n* [`accelerometer.getAcceleration()`(#f78183)](/docs/applab/accelerometer.getAcceleration/)- \r\n\r\n### Events\r\n\r\nThe accelerometer has a fair amount of event types as well. \r\n\r\n| Event |      \t\t\t\tDescription         |\r\n|-------|-------------------------------|\r\n| Change |Fired any time the accelerometer experiences a change |\r\n| Data | Fires every 50 milliseconds|\r\n| Single Tap | Fires only when a tap is detected next to the board. |\r\n| Double Tap | Fires only when two taps within x amount of time of each other is detected next to the board. |\r\n",
```
Comment (bethanyaconnor) [inline dashboard/config/programming_expressions/applab/accelerometer.json:6]: `accelerometer` links to itself here, but that's also what happens in CB so I decided not to go back and fix this.

File: dashboard/config/reference_guides/csp-2022/for-loops.json [hunk 2]
```diff
@@ -4,6 +4,6 @@
...
...n\nThe last piece of the for loop is the increment. It is the update piece of the loop. In order for the starting value of the variable to change we must update it each time through the loop. That is what the increment is used for. The increment is run at the end of each run through the for loop.  `i++ ` is just short hand for `i = i + 1`, which you might recognize as the counter pattern.\n\n## Example\n\nA for loop is similar to writing a bunch of if statements and counting as you go through each if statement. Let's take a look:\n\n### For Loop\n\n\tfor(var i = 0; i < 3; i++){\n    \tconsole.log(\"Hi!\");\n    }\n    \n### If Statements\n\nThis example does the same thing:\n\n\tvar i = 0;\n    if (i < 3){\n    \tconsole.log(\"Hi!\");\n        i++;\n    }\n    if (i < 3){\n    \tconsole.log(\"Hi!\");\n        i++;\n    }\n    if (i < 3){\n    \tconsole.log(\"Hi!\");\n        i++;\n    }\n    if (i < 3){\n    \tconsole.log(\"Hi!\");\n        i++;\n    }\n    \nThe 4th if statement here will not even run. In fact you could have if statements going on forever. It would only run the `console.log()` inside the if statement 3 times. When it checks the fourth if the condition will fail.",
```
Comment (bethanyaconnor) [inline dashboard/config/reference_guides/csp-2022/for-loops.json:7]: Follow up: Elijah had to go back through and update `console.log` links manually in #39983. I'll either update these in a separate pass or generate a list and pass it off to @tess323 to find an owner.

Comment (tim-dot-org) [reply dashboard/config/reference_guides/csp-2022/for-loops.json:7]: maybe listing all the places where `query.count != 1` in the script would be a good starting place for cleanup areas

**Review by tim-dot-org — APPROVED** [review]
script looks good and spot check of the ref guides seems fine!

**Pull request merged**